src/Centreon/Domain/Engine/EngineConfiguration.php

```diff
@@ -131,4 +131,28 @@ public static function removeIllegalCharacters(string $stringToAnalyse, ?string
         $illegalCharacters = html_entity_decode($illegalCharacters);
         return str_replace(str_split($illegalCharacters), '', $stringToAnalyse);
     }
+
+    /**
+     * Find all illegal characters from the given string.
+     *
+     * @param string $stringToAnalyse String for which we want to remove illegal characters
+     * @param string|null $illegalCharacters String containing illegal characters
+     * @return bool Return true if illegal characters have been found
+     */
+    public static function hasIllegalCharacters(string $stringToAnalyse, ?string $illegalCharacters): bool
+    {
+        if (null === $illegalCharacters) {
+            return false;
+        }
+        $illegalCharactersList = str_split(html_entity_decode($illegalCharacters), 1);
+        // Add space to the list as, space is not compliant with the RFC
```

Comment:
Warning, the user is not aware of this subtlety because space is not part of the unauthorized characters ("Illegal Object Name Characters").

Reply:
Yes indeed. But on the IHM, spaces are replaced by underscore for the user. Here we cannot replace something for the user.
Space is not compliant with the RFC and is not allowed in the specs... that's why I throw a specific message :
https://github.com/centreon/centreon/pull/9082/files#diff-34314464cea3309e49c82e38c9537a9dR15601

Reply:
In this case the space should be added in the list of illegal characters